src/AppBundle/Controller/Post/PostController.php

```diff
@@ -0,0 +1,194 @@
+<?php
+
+namespace AppBundle\Controller\Post;
+
+use Symfony\Bundle\FrameworkBundle\Controller\Controller;
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\Route;
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\Method;
+use AppBundle\Entity\Post;
+use Symfony\Component\VarDumper\Cloner\Data;
+use Symfony\Component\HttpFoundation\Request;
+
+class PostController extends Controller
+{
+    /**
+     * @Route("/post/{id}", requirements={"id" = "\d+"}, defaults={"id" =0}, name="postPage")
+     * @Method({"GET"})
+     *
+     * @param int $id
+     *
+     * @return object
+     */
+    public function showPostAction($id)
+    {
+        $categories = $this->getDoctrine()
+            ->getRepository('AppBundle\\Entity\\Category\\Category')
+            ->findAll();
```

Comment:
i think using paramconverter preferably. Than u can do like so $post->getCategories etc.  instead of all your extra queries